ledger/complete/mtrie/trie/trie.go

```diff
@@ -298,7 +299,14 @@ func update(
 	if lChild == lchildParent && rChild == rchildParent {
 		return parentNode
 	}
-	return node.NewInterimNode(nodeHeight, lChild, rChild)
+
+	n := node.NewInterimNode(nodeHeight, lChild, rChild)
+
+	if prune {
+		n, _ = n.Compactify()
+	}
+
+	return n
```

Comment:
I think this section should be a bit different, but I didn't think of the right way to write it yet. 
The issue is that the storage exhaustion attack mitigation might shortcut the pruning, i.e, it returns the same parent without pruning it.

Reply:
This seems fine if we assume the input tree is already totally pruned.